Docker is no longer able to execute 'nuxt build' - Unable to update lock within the stale threshold

### Version

[v2.5.0](https://github.com/nuxt.js/releases/tag/v2.5.0)

### Reproduction link

[https://github.com/ibuchan72390/docker-nuxt-hmr](https://github.com/ibuchan72390/docker-nuxt-hmr)

### Steps to reproduce

Attached is a simple nuxt repo I had originally used to demonstrate HMR viability in a nuxt dev instance housed in a docker container.  After updating Nuxt to 2.5.0, I'm no longer able to properly build out this container. 

Navigate into the node folder and execute the following from the CLI:
docker build -f test.dockerfile .

### What is expected ?

I would expect a lock can be gained on the targeted file and that we could successfully complete "nuxt build" within a docker container.

### What is actually happening?

Nuxt is failing to get a proper lock on the required files and it ends up failing the build entirely.  As such, I'm no longer able to build a production mode docker container for a nuxt application.

### Additional comments?

NPM Err:
/usr/app/node_modules/proper-lockfile/lib/lockfile.js:198
        onCompromised: (err) => { throw err; },
                                  ^

Error: Unable to update lock within the stale threshold
    at options.fs.stat (/usr/app/node_modules/proper-lockfile/lib/lockfile.js:121:25)
    at /usr/app/node_modules/graceful-fs/polyfills.js:285:20
    at FSReqWrap.oncomplete (fs.js:159:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! docker-test-dev@1.0.0 build: `nuxt build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the docker-test-dev@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-03-21T19_31_23_334Z-debug.log
The command '/bin/sh -c npm run build' returned a non-zero code: 

<!--cmty-->
<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8886">#c8886</a>)</em></sub></div>

--- a/packages/utils/src/locking.js
+++ b/packages/utils/src/locking.js
@@ -40,6 +40,7 @@ export async function lock({ id, dir, root, options }) {
     consola.fatal(`A lock with id '${id}' already exists on ${dir}`)
   }
 
+  options = getLockOptions(options)
   const release = await properlock.lock(lockPath, options)
 
   if (!release) {

--- a/packages/utils/test/locking.test.js
+++ b/packages/utils/test/locking.test.js
@@ -18,16 +18,16 @@ describe('util: locking', () => {
   beforeEach(() => jest.resetAllMocks())
   beforeEach(() => lockPaths.clear())
 
-  test('onCompromised lock is warn error by default', () => {
+  test('onCompromised lock warns on compromise by default', () => {
     defaultLockOptions.onCompromised()
     expect(consola.warn).toHaveBeenCalledTimes(1)
   })
 
   test('can override default options', () => {
-    const options = getLockOptions({ onCompromised: err => consola.warn(err) })
+    const options = getLockOptions({ onCompromised: err => consola.fatal(err) })
     options.onCompromised()
 
-    expect(consola.warn).toHaveBeenCalledTimes(1)
+    expect(consola.fatal).toHaveBeenCalledTimes(1)
   })
 
   test('createLockPath creates the same lockPath for identical locks', () => {
@@ -142,5 +142,18 @@ describe('util: locking', () => {
 
     expect(fs.removeSync).toHaveBeenCalledWith(path1)
     expect(fs.removeSync).toHaveBeenCalledWith(path2)
+  })
+
+  test('lock uses setLockOptions to set defaults', async () => {
+    const spy = properlock.lock.mockImplementationOnce(() => true)
+
+    await lock(lockConfig)
+
+    expect(spy).toHaveBeenCalledWith(expect.any(String), expect.any(Object))
+    const options = spy.mock.calls[0][1]
+    expect(options.stale).toBeDefined()
+    expect(options.onCompromised).toBeDefined()
+    expect(() => options.onCompromised()).not.toThrow()
+    expect(consola.fatal).not.toHaveBeenCalled()
   })
 })

